src/utils/process_info.rs

```diff
@@ -10,45 +8,31 @@ const TARGET_PROCESS: &str = "LeagueClientUx.";
 #[cfg(target_os = "macos")]
 const TARGET_PROCESS: &str = "LeagueClientUx";
 
-pub fn find_process(system: &System) -> Result<String, &'static str> {
-    system
+pub fn get_auth_info() -> Result<(String, u16), &'static str> {
+    let mut sys = System::new_all();
+    sys.refresh_processes();
+
+    let args = sys
         .processes()
         .values()
-        .find(|process| process.name() == TARGET_PROCESS)
-        .map(|process| process.cmd().join(" "))
-        .ok_or("Could not find a running LCU process!")
-}
-
-lazy_static! {
-    static ref PORT_RE: Regex = Regex::new(r"--app-port=([0-9]*)").unwrap();
-    static ref TOKEN_RE: Regex = Regex::new(r"--remoting-auth-token=([\w-]*)").unwrap();
-}
+        .find(|p| p.name() == TARGET_PROCESS)
+        .map(|p| p.cmd())
+        .ok_or("no LoL process")?;
```

Comment:
If the error messages are going to be changed, it makes sense to use an error enum instead of a new &str